### ISSUE TITLE
Answers statistics are wrong

### ISSUE THREAD

**Issue by idansc** [A2Zadeh/Social-IQ#1]
The paper and github mentions:

> Answer Statistics: Social-IQ contains a total of 30,000 correct (4 per question) and 22,500 (3 per question) incorrect answers. 

This is not the case for many of the questions. The fact there aren't exactly 3 incorrect answers also makes the multiple-choice variant impossible to evaluate.

**Comment by A2Zadeh** [comment]
Hello @idansc. That may be true for some of the questions and answers, but should be a small portion (I wonder why you are getting a large portion? Are you using the actual raw files? Can you tell me how many? Maybe I did something wrong in the release, so please let me know the number. Some videos have more than 6 questions but at least 6 questions have 4 correct and 3 incorrect in those videos.). In the final release after publication, we did one final evaluation of the questions and answers and removed some answers that were not following the Social-IQ rules (mostly incorrect answers that were not relevant to the questions or video, or correct answers that were repetitive). The same is also true for some of the transcriptions where the quality was not >90% (unfortunately, even manual transcriptions sometime are not perfectly done).  The missing transcriptions are going to be replaced with correct ones soon (you can expect a small improvement on the models afterwards). The answers are also pending re-annotations for the missing ones. We will update those as soon as possible. Both will be done in the next two weeks. The computational sequences should help build models for now. The updated ones will be identical in format with those.

**Comment by A2Zadeh** [comment]
@idansc And another clarification, if we have more than 3 correct or 4 incorrect, we only use the first 3 and last 4. They are essentially based on db dump which is based on time. I actually can fix that issue to avoid future confusions. We thought >4 incorrects and >3 corrects may facilitate some generation tasks but in reality I can see that it can also causes confusion.

**Comment by idansc** [comment]
Thanks Amir, I'm working with the raw data file.
First the number of answers with valid questions as you described are approx. 45K out of 63K. Please let me know if this is not expected. 

Given your answer ,I'm trying to align the raw file with the preprocessed file. Some Issues:
-  the number of videos reported are 1250, but in the raw and preprocessed file has 1016 videos.

- The preprocessed BERT features dimensions are (888, 6, 12, 3, 25, 768). My interpretation is the follow: 
1. 888 - Number of videos in train
2. 6 - Number of questions
3. 12 - not sure,is it related to mel features?
4. 3 - question,correct_ans,incorrect_ans
5. 25 - Length of sentence
6. 768 - bert's embedding dimension
what dimension represents the set of possible answers? 

Because the baseline works only with the preprocessed files,  a clarification is needed to have fair comparison with the baseline. 

Thanks again, really appreciate your hard work on this great dataset.

**Comment by A2Zadeh** [comment]
@idansc, Yes you are correct. We just reprocessed the files to only contain the questions and answers that are valid. The reason we have more is I included the questions that were also removed during validation set (which as you pointed out there is around 10K that is removed). Thought that might help, but now I don't see how it would. Long story short, I will update the question files tonight and re-zip the dataset so you can download it again, and all questions will have 4 correct and 3 incorrect answers. 

1016 is due to the fact that we have a private test set. The rest is there. We are also releasing a public test set. I will release the public test tonight with the questions. 

888 - Number of videos in train
6 - Number of questions
12 - this is possible binary combinations (pick one from correct, one from incorrect). So each question can be seen with one correct and one incorrect answer
3 - question,correct_ans,incorrect_ans (this is binary)
25 - Length of sentence
768 - bert's embedding dimension (last BERT layer)
what dimension represents the set of possible answers?

Of course, happy to answer any questions regarding the baseline. Does it work for you off the shelf? You should get around 65% on the validation set hopefully using the code.

Of course, glad we could deliver this dataset to you and the community. Social-IQ is sort of an umbrella term which will encompass multiple datasets that are being built right now (e.g. small subtasks related to social intelligence). Will let you know when they come out. 

Please keep me in the loop and don't hesitate to ask any questions.

**Comment by idansc** [comment]
Thanks Amir, I do get 65%. Although, we couldn't reproduce the ablation results. For instance, Q+A gives us more than 60%, and not 57% as you reported. Can you elaborate a bit more about this variant? We want to make sure we are not missing anything.

**Comment by A2Zadeh** [comment]
@idansc Great that you get 65%. The results in the paper are actually on the challenge test set. The challenge test set is slightly more balanced than the dev set. I don't know what is exactly in the challenge test set, but what I realized is that multimodal models generalize better to that set. So what I saw is that 65% of the multimodal model does almost get the same accuracy on test, but the model that uses say just language seems to overfit a bit on biases which do not generalize to the challenge test. Whenever your model is ready for challenge set, I can give you the data.

**Comment by A2Zadeh** [comment]
@idansc Just an update on this. It took us a bit longer to double check everything for removing the extra annotations. I will put them online soon (hopefully early in the weekend). Do you think there is any value including the extra annotations in a separate folder? Would you foresee a situation it could help your research?

**Comment by idansc** [comment]
Thanks Amir, waiting for the final version. I do believe the extra annotations can be useful. For instance, if we choose to evaluate on a generation variant.

**Comment by A2Zadeh** [comment]
@idansc Sorry this fell through the cracks. Great. I think I kept that information in the zip. I have been very busy in the past two weeks with AAAI, but will sent you the public test as soon as possible.

**Comment by A2Zadeh** [comment]
@idansc We just release the public test here: http://immortal.multicomp.cs.cmu.edu/raw_datasets/SocialIQ_PUBLIC_TEST.zip

I am closing this issue for now, feel free to ask any questions.
